### DISCUSSION TITLE
Nested Field Array adding new value.

### DISCUSSION THREAD

**Discussion by yjp20** [react-hook-form/react-hook-form#2185] [Q&A] [answered]
Hello, as my title describes, I have a problem with using nested field arrays.

There's no problems when I'm trying to add a single item to the list, or getting the values off of the default values. However, when I try to add an object like 
```
{
  id: "hello",
  values: [
    "a",
    "b",  
  ],
}
```
I find it difficult to implement because you can only add one at a time with append and you can't call an action more than once. I reckon there's some hacky workarounds using useEffect but I'm not sure if that's ideal in this case.

**Comment by bluebill1049** [comment, marked as answer]
can you share a simple codesandbox?

**Reply by yjp20** [reply]
https://codesandbox.io/s/affectionate-https-lwbl8?file=/src/App.js

**Reply by bluebill1049** [reply]
actually this codesandbox maybe helpful: https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-x7btr

**Reply by yjp20** [reply]
I've seen that one, but it doesn't explain how to add parts of both the larger array and nested array at the same time.
Neither does it explain how to append/prepend multiple elements at once.

**Reply by bluebill1049** [reply]
it almost sounds like you want to use reset the whole `FieldArray` when you want to update multiple field arrays. 

in terms of append multiple, you can use the array argument for  append([]) and prepend([])

**Reply by yjp20** [reply]
I'm not sure what you mean by reset the whole FieldArray, but given that you could append multiple I think I can make a workaround.

I did not know that you could append using an array argument, that's really helpful. I'm not sure if it's in the docs -- i've looked for quite a while and haven't found it, but I may be wrong. It might be a nice thing to add if it doesn't exist.

**Reply by bluebill1049** [reply]
I mean you can reset the whole FieldArray as well.

`useEffect(() => { reset({test: [xxx]}) }, [reset])`

**Reply by bluebill1049** [reply]
> I'm not sure what you mean by reset the whole FieldArray, but given that you could append multiple I think I can make a workaround.
> 
> I did not know that you could append using an array argument, that's really helpful. I'm not sure if it's in the docs -- i've looked for quite a while and haven't found it, but I may be wrong. It might be a nice thing to add if it doesn't exist.

Yea, i think the append with array is not clear in the doc, will get that updated.

**Reply by yjp20** [reply]
Wow the first method works great. Also you are about the most responsive and dedicated maintainer I've ever seen.

**Comment by bluebill1049** [comment]
hey @yjp20, I just saw the sponsorship from you, sorry about the late notice, too many GitHub emails. I am really appreciated your support. Do you have a twitter account that I can follow and connect?

**Comment by yjp20** [comment]
no, but I wanted to support people like you who spend good time to make my
job a lot easier. have a coffee or something :)

**Reply by bluebill1049** [reply]
Thank you @yjp20 🙏
